eth/gasprice/gasprice_test.go

```diff
@@ -100,12 +100,7 @@ func newTestBackend(t *testing.T, londonBlock *big.Int, pending bool) *testBacke
 		}
 		signer = types.LatestSigner(gspec.Config)
 	)
-	if londonBlock != nil {
-		gspec.Config.LondonBlock = londonBlock
-		signer = types.LatestSigner(gspec.Config)
-	} else {
-		gspec.Config.LondonBlock = nil
-	}
+	gspec.Config.LondonBlock = londonBlock
```

Comment:
I also found this non-obvious in the original PR where this conditional got introduced. 
With your change, the assignment of `signer` gets lost. This changes the outcome
because `LatestSigner` may choose another signer type if London is enabled.

Both your version and the original version have another issue: `params.TestChainConfig`
is modified without copying it first.

Reply:
Ah I didn't realize this. Would it make sense to copy the config, modify the LondonBlock, and then create the signer to remove the no-op here?

Reply:
Yes, that would be better.